sky/shell/ui/engine.h

```diff
@@ -94,6 +96,8 @@ class Engine : public UIDelegate,
   gfx::Size physical_size_;
   std::string initial_route_;
   blink::SkyDisplayMetrics display_metrics_;
+  std::string language_code_;
+  std::string country_code_;
```

Comment:
Whatever code holds the cache should probably do a difference check before forwarding the notification. I think the way the code is currently structured, you'll dispatch an update when an Android device is rotated.